variables.tf

```diff
@@ -64,6 +64,20 @@ variable "kms_encryption_enabled" {
   type        = bool
   description = "Set this to true to control the encryption keys used to encrypt the data that you store for AppID. If set to false, the data is encrypted by using randomly generated keys. For more info on securing data in AppID, see https://cloud.ibm.com/docs/appid?topic=appid-mng-data"
   default     = false
+  validation {
+    condition     = var.kms_encryption_enabled && var.plan != "graduated-tier" ? false : true
+    error_message = "kms encryption is only supported for graduated-tier plan"
+  }
+
+  validation {
+    condition     = !var.kms_encryption_enabled && (var.existing_kms_instance_guid != null || var.kms_key_crn != null) ? false : true
+    error_message = "When passing values for var.existing_kms_instance_guid or/and var.kms_key_crn, you must set var.kms_encryption_enabled to true. Otherwise unset them to use default encryption"
+  }
+
+  validation {
+    condition     = var.kms_encryption_enabled && (var.existing_kms_instance_guid == null || var.kms_key_crn == null) ? false : true
+    error_message = "When setting var.kms_encryption_enabled to true, a value must be passed for var.existing_kms_instance_guid and var.kms_key_crn"
```

Comment:
A key cannot exist without an associated instance. Therefore, if `kms_encryption_enabled` is set to `true`, a `kms_key_crn` must be provided. You do not need a condition here for `existing_kms_instance_guid`.  You can add another validation in the `kms_key_crn` variable to have the `existing_kms_instance_guid` value passed.